### PR TITLE
Staging Sites: Display error when production->staging synchronization fails

### PR DESCRIPTION
**Pull request by wojtekn**
Related to https://github.com/Automattic/dotcom-forge/issues/4271

## Proposed Changes

In this PR, I propose to display an error when production<->staging synchronization fails:

Screenshot 2023-10-27 at 15 50 04

## Testing Instructions

1. Create a site with a Business plan
2. Navigate to Settings -> Hosting Configuration
3. Add a staging site
4. Using `wpsh`, fill meta to simulate failed synchronization:
```
update_site_meta( PRODUCTION_BLOG_ID, 'site_staging_sync_data', array( 'status' => 'failed') )
```
5. Refresh Hosting Configuration
6. Confirm that error is displayed with CTA asking customer to contact support

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/hosting/staging-site-card/card-content/staging-sync-card.tsx [hunk 1]
```diff
@@ -19,15 +19,23 @@ import { useCheckSyncStatus } from '../use-site-sync-status';
...
 		return false;
 	}
-	return Object.values( STAGING_SYNC_ERROR_CODES ).includes( error );
+	return Object.values( STAGING_SYNC_JETPACK_ERROR_CODES ).includes( error );
+}
+function useIsFailedSyncError( error: string | null | undefined ) {
+	if ( ! error ) {
+		return false;
+	}
+	return Object.values( STAGING_SYNC_FAILED_ERROR_CODES ).includes( error );
```
Comment (kozer) [inline client/my-sites/hosting/staging-site-card/card-content/staging-sync-card.tsx]: `STAGING_SYNC_FAILED_ERROR_CODES` is an Array, not an object.
The code you wrote works, but the array object already has an `includes` [function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes), so we can just write
```suggestion
	return STAGING_SYNC_FAILED_ERROR_CODES.includes( error );
```

File: client/my-sites/hosting/staging-site-card/card-content/staging-sync-card.tsx [hunk 2]
```diff
@@ -19,15 +19,23 @@ import { useCheckSyncStatus } from '../use-site-sync-status';
...
 ];
-function useIsPermanentSyncError( error: string | null | undefined ) {
+const STAGING_SYNC_FAILED_ERROR_CODES = [ 'staging_site_sync_failed' ];
+
+function useIsJetpackConnectionSyncError( error: string | null | undefined ) {
 	if ( ! error ) {
 		return false;
 	}
-	return Object.values( STAGING_SYNC_ERROR_CODES ).includes( error );
+	return Object.values( STAGING_SYNC_JETPACK_ERROR_CODES ).includes( error );
```
Comment (kozer) [inline client/my-sites/hosting/staging-site-card/card-content/staging-sync-card.tsx]:
```suggestion
	return STAGING_SYNC_JETPACK_ERROR_CODES.includes( error );
```

**Review by kozer — CHANGES_REQUESTED** [review]

**Comment by kozer** [comment]
I checked it and it seems to work as expected! I have added some code suggestions though!

However, when we refresh the pager, and before the data gets loaded, there is a timeframe when the checkboxes appear.
Can we `fetchingStatus` in order to avoid that?

We can also create another ticket for that and implement a "skeleton loader" as well so this will make things more complete from a UI/UX perspective.

**Comment by wojtekn** [comment]
> I checked it and it seems to work as expected! I have added some code suggestions though!

Thanks for testing it. I applied the suggestion.

> We can also create another ticket for that and implement a "skeleton loader" as well so this will make things more complete from a UI/UX perspective.

Would you mind adding a new issue for that for our maintenance backlog?

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/9565242

Some locales (Hebrew) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @wojtekn for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/9565242).

**Pull request merged**